# Importing Libraries

In [2]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# Load the data

In [3]:
df=pd.read_csv('Age_Height.csv')

In [4]:
df.head()

Age  Height
0   10     138
1   11     138
2   12     138
3   13     139
4   14     139

In [5]:
df.tail()

Age  Height
66   76     204
67   77     205
68   78     206
69   79     207
70   80     208

In [6]:
df.shape

(71, 2)

Splitting Dependent and independent data

In [10]:
x = df.iloc[:,0].values
y = df.iloc[:,1].values

In [14]:
x = x.reshape(-1,1)
y = y.reshape(-1,1)

# Data Preprocessing

In [15]:
from sklearn.model_selection import train_test_split

X_train,X_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=30)

# Model Training and Prediction

In [16]:
from sklearn.tree import DecisionTreeRegressor

model=DecisionTreeRegressor()
model.fit(X_train,y_train)

DecisionTreeRegressor()

In [17]:
y_pred=model.predict(X_test)

In [18]:
y_pred

array([144., 168., 159., 141., 146., 185., 206., 172., 163., 180., 144.,
       190., 147., 150., 195.])

# Model Evaluation

In [20]:
from sklearn.metrics import r2_score

print('R2 score for Decision Tree Regression:',r2_score(y_test,y_pred))

R2 score for Decision Tree Regression: 0.9924455205811138


# Sample Testing

In [21]:
df.sample(1)

Age  Height
11   21     141

In [22]:
model.predict([[21]])

array([141.])

In [25]:
df.sample(2)

Age  Height
14   24     142
20   30     145

In [26]:
model.predict([[24],[30]])

array([142., 144.])

The R2 score of 0.992 reflects an excellent fit of the model to the test data, indicating its ability to capture the relationship between age and height.
When predicting a height for an age of 64, the model estimated 172 cm.
When predicting a height for an age of 53, the prediction was 163 cm.
